### PR TITLE
Mepts 652 apss a1 indicator

### PR DESCRIPTION
**Pull request by cfaife**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/AgeOnReportEndDateDateCalculation.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import org.openmrs.calculation.result.CalculationResultMap;
+import org.openmrs.module.eptsreports.reporting.calculation.AbstractPatientCalculation;
+import org.openmrs.module.eptsreports.reporting.calculation.BooleanResult;
+import org.openmrs.module.eptsreports.reporting.utils.EptsCalculationUtils;
+import org.openmrs.module.reporting.common.Birthdate;
+import org.openmrs.module.reporting.data.person.definition.BirthdateDataDefinition;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AgeOnReportEndDateDateCalculation extends AbstractPatientCalculation {
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/AgeOnReportEndDateDateCalculation.java:25]: @cfaife can you please put some javadocs on the propose of this calculation.... will be helpful tomorrow

Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/AgeOnReportEndDateDateCalculation.java:25]: Ok sure I will

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/AgeOnReportEndDateDateCalculation.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      PatientCalculationContext context) {
+    CalculationResultMap map = new CalculationResultMap();
+
+    CalculationResultMap birthDates =
+        EptsCalculationUtils.evaluateWithReporting(
+            new BirthdateDataDefinition(), cohort, null, null, context);
+    Integer minAge = (Integer) parameterValues.get(MIN_AGE);
+    Integer maxAge = (Integer) parameterValues.get(MAX_AGE);
+    Date endDate = (Date) context.getFromCache(ON_OR_BEFORE);
+    Boolean considerPatientThatStartedBeforeWasBorn =
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/AgeOnReportEndDateDateCalculation.java]: No need for this...

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/APSSResumoTrimestralCohortQueries.java [hunk 3]
```diff
@@ -72,14 +124,46 @@ public CohortDefinition getA1() {
...
+   *             ${patientFoundYesConcept})
+   *         <li>And “encounter_datetime” Between StartDate and EndDate
+   *       </ul>
+   * </ul>
+   *
    * @return {@link CohortDefinition}
    */
   public CohortDefinition getB1() {
-    SqlCohortDefinition sqlCohortDefinition = new SqlCohortDefinition();
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/APSSResumoTrimestralCohortQueries.java]: I know there's a relation between this 2 but we need to see how have stuff merged while they need to be reviewed

**Review by psmatsinhe — COMMENTED** [review]
Please see my comments

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/AgeOnReportEndDateDateCalculation.java (hunk 2 above)
Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/AgeOnReportEndDateDateCalculation.java]: yes removed already

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/AgeOnReportEndDateDateCalculation.java (hunk 1 above)
Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/AgeOnReportEndDateDateCalculation.java:25]: done

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/APSSResumoTrimestralCohortQueries.java (hunk 3 above)
Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/APSSResumoTrimestralCohortQueries.java]: yes sure

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/APSSResumoTrimestralCohortQueries.java [hunk 4]
```diff
@@ -240,4 +344,191 @@ public SqlCohortDefinition getAllPatientsRegisteredInEncounterFichaAPSSANDPP() {
...
+            + "     INNER JOIN obs o "
+            + "        ON o.encounter_id=e.encounter_id "
+            + "WHERE p.voided = 0 "
+            + "    AND e.voided = 0 "
+            + "    AND o.voided = 0 "
+            + "    AND e.encounter_type = ${prevencaoPositivaSeguimentoEncounterType} "
+            + "    AND (o.concept_id = ${pp1Concept} AND o.value_coded = ${patientFoundYesConcept}) "
+            + "    AND encounter_datetime "
+            + "        BETWEEN :startDate AND :endDate "
+            + "    AND e.location_id = :location "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/APSSResumoTrimestralCohortQueries.java]: Can't you put this query as with concept_id in (pp1,pp2....) and o.value_code=$patientfoundyes?

**Review by psmatsinhe — CHANGES_REQUESTED** [review]

**Review by psmatsinhe — APPROVED** [review]
Looks good to me

**Pull request merged**